assets/app/scripts/directives/editRequestLimit.js

```diff
@@ -0,0 +1,234 @@
+              limitWithinRatio = true;
+
+          if (request && limit) {
+            // Limit must be greater than or equal to request.
+            limitLargerThanRequest = limit >= request;
+
+            // Limit must be within the max limit/request ratio if defined.
+            if (scope.maxLimitRequestRatio) {
+              limitWithinRatio = (limit / request) <= scope.maxLimitRequestRatio;
+            }
+          }
+
+          scope.form.amount.$setValidity('limitLargerThanRequest', limitLargerThanRequest);
+          scope.form.amount.$setValidity('limitWithinRatio', limitWithinRatio);
+        };
+
+        // Update view from model.
+        ngModel.$render = function() {
+          var update = _.spread(function(amount, unit) {
+            if (amount) {
+              scope.amount = Number(amount);
+              scope.unit = unit;
+              // If the unit already set in the resource isn't in the list, add it.
+              addUnitOption(unit);
+            } else {
+              scope.amount = null;
+            }
+          });
+          update(amountAndUnit(ngModel.$viewValue, scope.type));
+        };
+
+        // Update model from view.
+        scope.$watchGroup(['amount', 'unit'], function() {
+          validateLimitRange();
+          validateLimitAgainstRequest();
+          if (!scope.amount) {
+            ngModel.$setViewValue(undefined);
+          } else {
+            ngModel.$setViewValue(scope.amount + scope.unit);
+          }
+        });
+
+        scope.$watchGroup(['limitRangeMin', 'limitRangeMax'], validateLimitRange);
+        scope.$watch('request', validateLimitAgainstRequest);
+      }
+    };
+  })
+  .directive('editRequestLimit', function($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        resources: '=',
+        type: '@',
+        limitRanges: '='
+      },
+      templateUrl: 'views/_edit-request-limit.html',
+      link: function(scope) {
+        var usageValue = $filter('usageValue');
+        var isSmaller = function(candidate, previous) {
+          if (!candidate) {
+            return false;
+          }
+
+          // No previous value, so use candidate.
+          if (!previous) {
+            return true;
+          }
+
+          // Normalize units to compare.
+          return usageValue(candidate) < usageValue(previous);
+        };
+
+        var isLarger = function(candidate, previous) {
+          if (!candidate) {
+            return false;
+          }
+
+          // No previous value, so use candidate.
+          if (!previous) {
+            return true;
+          }
+
+          // Normalize units to compare.
+          return usageValue(candidate) > usageValue(previous);
+        };
+
+        var updateLimitRanges = function() {
+          scope.limits = {};
+          angular.forEach(scope.limitRanges, function(limitRange) {
+            angular.forEach(limitRange.spec.limits, function(limit) {
+              // Find the largest minimum.
```

Comment:
should this be in a LimitRanges service, LimitRanges.getEffectiveLimits() or something.  Feels like we are going to need this somewhere else.